nautobot/core/testing/api.py

```diff
@@ -214,24 +232,102 @@ def test_list_objects_anonymous(self):
-                "In order to test the brief API parameter the brief fields need to be manually added to "
-                "self.brief_fields. If this is already the case, perhaps the serializer is implemented incorrectly?",
-            )
+
+            response_data = response.data["results"][0]
+            for field in depth_fields:
+                self.assertEqual(field in response_data, True)
+                if isinstance(response_data[field], list):
+                    for entry in response_data[field]:
+                        try:
+                            uuid.UUID(str(entry))
+                        except ValueError:
+                            self.fail(f"Response data field {field} contains invalid UUID")
+                else:
+                    if response_data[field] is not None:
+                        try:
+                            uuid.UUID(str(response_data[field]))
+                        except ValueError:
+                            self.fail(f"Response data field {field} contains invalid UUID")
+
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+        def test_list_objects_depth_1(self):
+            """
+            GET a list of objects using the "?depth=1" parameter.
+            """
+            depth_fields = self.get_depth_fields()
+            self.add_permissions(f"{self.model._meta.app_label}.view_{self.model._meta.model_name}")
+            url = f"{self._get_list_url()}?depth=1"
+            response = self.client.get(url, **self.header)
+
+            self.assertHttpStatus(response, status.HTTP_200_OK)
+            self.assertIsInstance(response.data, dict)
+            self.assertIn("results", response.data)
+            self.assertEqual(len(response.data["results"]), self._get_queryset().count())
+
+            response_data = response.data["results"][0]
+            for field in depth_fields:
+                self.assertEqual(field in response_data, True)
+                if isinstance(response_data[field], list):
+                    for entry in response_data[field]:
+                        try:
+                            uuid.UUID(str(entry["id"]))
+                        except ValueError:
+                            self.fail(f"Response data field {field} contains invalid UUID")
+                else:
+                    if response_data[field] is not None:
+                        try:
+                            uuid.UUID(str(response_data[field]["id"]))
+                        except ValueError:
+                            self.fail(f"Response data field {field} contains invalid UUID")
+
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=[])
+        def test_list_objects_depth_2(self):
+            """
+            GET a list of objects using the "?depth=2" parameter.
+            TreeModel Only
+            """
+            if not issubclass(self.model, TreeModel):
+                self.skipTest(f"Model {self.model._meta.model_name} is not a TreeModel")
```

Comment:
Can we instead make this another testcase mixin class and explicitly add it to the small number of tree-model tests?